text/0000-add-geo-processing-abilities-to-skylark.md

```diff
@@ -0,0 +1,535 @@
+- Feature Name: Add geo processing abilities to starlark 
+- Start Date: 2018-10-02
+- RFC PR: <!-- (leave this empty) -->
+- Issue: <!-- (leave this empty) -->
+
+# Summary
+[summary]: #summary
+
+
+# Motivation
+[motivation]: #motivation
+
+A lot of open public data, environmental data, government data etc has some kind of geospatial properties. Being able to 
+perform geospatial operations on these datasets would be a huge win for qri and could help increase adoption of qri as a means
+to distribute these types of dataset
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+When working with spatial datasets, those associated with a point, line, polygon etc we often need to be able to perform manipulations
+of the data based on those geometries. The Geo module of Starlib provides the basic data types and methods to be able to do just this.
+Based on the python [Shapley](https://pypi.org/project/Shapely/) library the Geo module allows you to perform operations such as 
+
+- Load Points Lines and Polygons in GeoJSON format and represent these as types within qri
+- Calculate distances between geometric features 
+- Perform filtering based on geometric conditions such as Within, Intersects, Contains etc 
+- Manipulate lines and polygons with operations like intersects, union etc.
+
+Some use-cases where this would be useful would be 
+
+- Working with 311 complaint data from a open government data portal and wanting to produce a summary of counts of complaint by type in each census tract. This would require grabbing data from the dataset but also the census tracts for an area and performing a spatial join and aggregation on that data.
+- Augmenting a dataset of bird sightings with the distance to the nearest wetlands.
+- Calculating the overlap of flood regions and building footprints within a region to assess risk
+
+An example of this which takes a dataset of 311 complaints, downloads the New York borough boundaries form the data portal and then counts how many complaints are in each borough, might look like this:
+
+```python 
+load("qri.sky", "qri")
```

Comment:
quick heads-up that we'll be making the formal transition to the "starlark" name in the very-near future. I've talked it over with Alan Donovan, new file extension will be `.star`.